File: backend-nodejs/src/Models/DTO/EventDTO.js
class EventDTO {

    constructor(
        eventId,
        venueDTO,
        eventTypeDTO,
        eventDescription,
        eventName,
        eventImageUrl,
        startDate,
        endDate,
        ticketCategoriesDTOList
    ) {
        this.eventId = eventId;
        this.venue = venueDTO;
        this.eventType = eventTypeDTO;
        this.eventDescription = eventDescription;
        this.eventName = eventName;
        this.eventImageUrl = eventImageUrl;
        this.startDate = startDate;
        this.endDate = endDate;
        this.ticketCategories = ticketCategoriesDTOList;
    }

}

module.exports = EventDTO;

File: backend-dotnet/EventTicketSystem/EventTicketSystem/Models/DTOs/EventDTO.cs
namespace EventTicketSystem.Models.DTOs
{
    public class EventDTO
    {
        public long EventId { get; set; }

        public VenueDTO? Venue { get; set; }

        public EventTypeDTO? EventType { get; set; }

        public string? EventDescription { get; set; }

        public string? EventName { get; set; }

        public string? EventImageUrl { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public ICollection<TicketCategoryDTO> TicketCategories { get; set; } = new List<TicketCategoryDTO>();


    }
}
